src/frontend/types/yaksa_blkindx.c

```diff
@@ -131,11 +133,11 @@ int yaksa_create_indexed_block(int count, int blocklength, const int *array_of_d
     rc = yaksi_create_hindexed_block(count, blocklength, real_array_of_displs, intype, &outtype);
     YAKSU_ERR_CHECK(rc, fn_fail);
 
-    free(real_array_of_displs);
-
     *newtype = outtype->id;
 
   fn_exit:
+    if (real_array_of_displs)
```

Comment:
do you need this check?

Reply:
Yeah, in the error case, we might end up here before the array is actually allocated.

Reply:
`free(NULL)` is safe right?

Reply:
No, but we check for that case when freeing the type.